data-node/src/main/java/org/graylog/datanode/bootstrap/DatanodeBootstrap.java

```diff
@@ -34,89 +32,35 @@
 import org.graylog2.plugin.Plugin;
 import org.graylog2.plugin.Tools;
-import org.graylog2.plugin.cluster.ClusterConfigService;
-import org.graylog2.shared.bindings.FreshInstallDetectionModule;
 import org.graylog2.shared.bindings.IsDevelopmentBindings;
-import org.graylog2.shared.plugins.ChainingClassLoader;
 import org.graylog2.shared.system.activities.Activity;
 import org.graylog2.shared.system.activities.ActivityWriter;
 import org.jsoftbiz.utils.OS;
 import org.slf4j.Logger;
 import org.slf4j.LoggerFactory;
 
-import java.nio.charset.StandardCharsets;
-import java.nio.file.Files;
-import java.nio.file.LinkOption;
-import java.nio.file.Path;
-import java.nio.file.Paths;
-import java.nio.file.StandardOpenOption;
 import java.util.Collection;
 import java.util.List;
 import java.util.Set;
 import java.util.concurrent.TimeUnit;
 import java.util.concurrent.TimeoutException;
 import java.util.stream.Collectors;
 
-import static com.google.common.base.Strings.isNullOrEmpty;
-
-public abstract class ServerBootstrap extends CmdLineTool {
-    private static final Logger LOG = LoggerFactory.getLogger(ServerBootstrap.class);
-    private boolean isFreshInstallation;
+public abstract class DatanodeBootstrap extends DatanodeCmdLineTool {
+    private static final Logger LOG = LoggerFactory.getLogger(DatanodeBootstrap.class);
     protected Configuration configuration;
 
-    protected ServerBootstrap(String commandName, Configuration configuration) {
+    protected DatanodeBootstrap(String commandName, Configuration configuration) {
         super(commandName);
         this.commandName = commandName;
         this.configuration = configuration;
     }
 
-    @Option(name = {"-p", "--pidfile"}, description = "File containing the PID of Graylog DataNode")
-    private String pidFile = TMPDIR + FILE_SEPARATOR + "datanode.pid";
-
-    @Option(name = {"-np", "--no-pid-file"}, description = "Do not write a PID file (overrides -p/--pidfile)")
-    private boolean noPidFile = false;
-
     protected abstract void startNodeRegistration(Injector injector);
 
-    public String getPidFile() {
-        return pidFile;
-    }
-
-    public boolean isNoPidFile() {
-        return noPidFile;
-    }
-
-    private boolean isFreshInstallation() {
-        return isFreshInstallation;
-    }
-
-    private void registerFreshInstallation() {
-        this.isFreshInstallation = true;
-    }
-
-    @Override
-    protected void beforeStart(TLSProtocolsConfiguration tlsProtocolsConfiguration, Configuration configuration) {
-        super.beforeStart(tlsProtocolsConfiguration, configuration);
-
-        // Do not use a PID file if the user requested not to
-        if (!isNoPidFile()) {
-            savePidFile(getPidFile());
-        }
-        // This needs to run before the first SSLContext is instantiated,
-        // because it sets up the default SSLAlgorithmConstraints
-        applySecuritySettings(tlsProtocolsConfiguration);
```

Comment:
Do we not need the security settings for Data Node?

Reply:
Ping.

Reply:
You're right. I moved that to the general `beforeStart` now since setting the values doesn't hurt for all kinds of nodes.